R/plot_point.r

```diff
@@ -221,15 +221,15 @@ plot_linear_scatter <- function(df, tooltip_data=NULL, gvis_filename=NULL, corme
     if (!is.null(color_low) | !is.null(color_high)) {
         ## If you want to color the above or below identity line points, then you will need subsets to define them
         tmpdf <- df
-        tmpdf[["ratio"]] <- tmpdf[, 2] / tmpdf[, 1]
+        tmpdf[["ratio"]] <- tmpdf[, 2] - tmpdf[, 1]
         subset_points <- suppressMessages(get_sig_genes(tmpdf, column="ratio", ...))
-        high_subset = subset_points[["up_genes"]]
-        low_subset = subset_points[["down_genes"]]
-        original_df = tmpdf
-        high_index = rownames(original_df) %in% rownames(high_subset)
-        high_df = original_df[high_index, ]
-        low_index = rownames(original_df) %in% rownames(low_subset)
-        low_df = original_df[low_index, ]
+        high_subset <- subset_points[["up_genes"]]
+        low_subset <- subset_points[["down_genes"]]
+        original_df <- tmpdf
+        high_index <- rownames(original_df) %in% rownames(high_subset)
+        high_df <- original_df[high_index, ]
+        low_index <- rownames(original_df) %in% rownames(low_subset)
+        low_df <- original_df[low_index, ]
         first_vs_second <- first_vs_second +
```

Comment:
Heh, far funnier is what I did to make 100% certain that the
annotation data matches the count data in an expressionset:

here is a snippet to make you laugh:
>     tmp_gene_info <- gene_info
>     tmp_gene_info[["temporary_id_number"]] <- 1:nrow(tmp_gene_info)
>     counts_and_annotations <- merge(all_count_matrix, tmp_gene_info, by="row.names", all.x=TRUE)
>     counts_and_annotations <- counts_and_annotations[order(counts_and_annotations[["temporary_id_number"]]), ]
>     final_annotations <- counts_and_annotations[, colnames(counts_and_annotations) %in% colnames(gene_info) ]
>     rownames(final_annotations) <- counts_and_annotations[["Row.names"]]
>     final_counts <- counts_and_annotations[, colnames(counts_and_annotations) %in% colnames(all_count_matrix) ]
>     rownames(final_counts) <- counts_and_annotations[["Row.names"]]
>     rm(counts_and_annotations)
>     rm(tmp_gene_info)

  It turns out that this is necessary for a reason which is utterly
surprising:  if you do an rpkm(expressionset), even through an
expressionset theoretically has the gene lengths in the annotation, you
must still pass it a named vector of gene lengths.  Ok, fine -- since it
is named one might assume that it will make sure to do
cpm(gene)/length(same_gene)  -- nope, it just passes the lengths in the
order received, as a result if your gene annotations are in a different
order than the count tables and you do not take that into account, then
you are fubar.  Ergo, forcing the order of the annotations and counts in
this admittedly stupid fashion is actually a good thing.


On 06/16/2016 01:43 PM, Keith Hughitt wrote:
> In R/plot_point.r
> <https://github.com/elsayed-lab/hpgltools/pull/48#discussion_r67390895>:
> 
>>          subset_points <- suppressMessages(get_sig_genes(tmpdf, column="ratio", ...))
>> -        high_subset = subset_points[["up_genes"]]
>> -        low_subset = subset_points[["down_genes"]]
>> -        original_df = tmpdf
>> -        high_index = rownames(original_df) %in% rownames(high_subset)
>> -        high_df = original_df[high_index, ]
>> -        low_index = rownames(original_df) %in% rownames(low_subset)
>> -        low_df = original_df[low_index, ]
>> +        high_subset <- subset_points[["up_genes"]]
>> +        low_subset <- subset_points[["down_genes"]]
>> +        original_df <- tmpdf
>> +        high_index <- rownames(original_df) %in% rownames(high_subset)
>> +        high_df <- original_df[high_index, ]
>> +        low_index <- rownames(original_df) %in% rownames(low_subset)
>> +        low_df <- original_df[low_index, ]
>>          first_vs_second <- first_vs_second +
> 
> way to bite the bullet ;)
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/elsayed-lab/hpgltools/pull/48/files/3727610e90045ce3eb9bdc174ccca89c95d4879c..c7a42728c2265a3d67736c4ac2c26a997e936b75#r67390895>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAEA7ZdVUHNlnRKJsSydhWJJoFYBQACDks5qMYtLgaJpZM4IxiuK>.
>